src/System Application/App/AI/src/Azure OpenAI/AOAIOperationResponse.Codeunit.al

```diff
@@ -57,6 +62,60 @@ codeunit 7770 "AOAI Operation Response"
         exit(Error);
     end;
 
+    /// <summary>
+    /// Get whether the operation was a function call.
+    /// </summary>
+    /// <returns>True if it was a function call, false otherwise.</returns>
+    procedure IsFunctionCall(): Boolean
+    begin
+        exit(FunctionCallingName <> '');
+    end;
+
+    /// <summary>
+    /// Get whether the function call was successful.
+    /// </summary>
+    /// <returns>True if the call was successful, false otherwise.</returns>
+    procedure IsFunctionCallSuccess(): Boolean
```

Comment:
I wonder if it would make sense to ask "GetFunctionCall" and return a codeunit for the function call itself instead of having one GOD response codeunit that handles everything? Then you can ask that whether it's a success, the name, values etc.

Reply:
Depends on how many codeunits you want to deal with. That is of course a possibility. 
I'll leave this comment open for others to comment on their thoughts. At this moment, I favor neither opinion.

Reply:
Ended up making the change. It is cleaner.